extension/src/popup/components/sendPayment/SendConfirm/SubmitResult/index.tsx

```diff
@@ -1,59 +1,140 @@
 import { ROUTES } from "popup/constants/routes";
 import {
   resetSubmission,
   transactionDataSelector,
+  transactionSubmissionSelector,
 } from "popup/ducks/transactionSubmission";
+import { IdenticonImg } from "popup/components/identicons/IdenticonImg";
+import {
+  AccountDoesntExistWarning,
+  shouldAccountDoesntExistWarning,
+} from "popup/components/sendPayment/SendTo";
+
 import "./styles.scss";
 
-// TODO - helper for asset names
 export const SubmitSuccess = ({ viewDetails }: { viewDetails: () => void }) => {
   const dispatch = useDispatch();
   const { destination, amount, asset } = useSelector(transactionDataSelector);
+
+  const horizonAsset = getAssetFromCanonical(asset);
+
   return (
-    <div className="SubmitSuccess">
-      <div>Successfuly sent</div>
-      <div>
-        {amount} {asset}
+    <div className="SubmitResult">
+      <div className="SubmitResult__header">Successfuly sent</div>
+      <div className="SubmitResult__amount">
+        {amount} {horizonAsset.code}
+      </div>
+      <div className="SubmitResult__icon">
+        <Icon.ArrowDownCircle />
+      </div>
+      <div className="SubmitResult__identicon">
+        <IdenticonImg publicKey={destination} />
+        <span>{truncatedPublicKey(destination)}</span>
+      </div>
+      <div className="SubmitResult__button-rows__success">
+        <Button fullWidth onClick={() => viewDetails()}>
+          Transaction Details
+        </Button>
+        <Button
+          fullWidth
+          variant={Button.variant.tertiary}
+          onClick={() => {
+            dispatch(resetSubmission());
+            navigateTo(ROUTES.account);
+          }}
+        >
+          Done
+        </Button>
       </div>
-      <div>{truncatedPublicKey(destination)}</div>
-      <Button fullWidth onClick={() => viewDetails()}>
-        Transaction Details
-      </Button>
-      <Button
-        variant={Button.variant.tertiary}
-        onClick={() => {
-          dispatch(resetSubmission());
-          navigateTo(ROUTES.account);
-        }}
-      >
-        Done
-      </Button>
     </div>
   );
 };
 
 export const SubmitFail = () => {
   const dispatch = useDispatch();
-  const { destination } = useSelector(transactionDataSelector);
+  const {
+    destinationBalances,
+    transactionData: { destination, amount, asset },
+  } = useSelector(transactionSubmissionSelector);
+
+  const horizonAsset = getAssetFromCanonical(asset);
+
+  const decideError = () => {
+    // unfunded destination
+    if (
+      shouldAccountDoesntExistWarning(
+        destinationBalances.isFunded || false,
+        asset,
+        amount,
+      )
+    ) {
+      return <AccountDoesntExistWarning />;
+    }
+
+    // no trustline
+    if (asset !== Asset.native().toString()) {
+      let found = false;
+      Object.keys(destinationBalances.balances || {}).forEach((key) => {
+        if (asset === key) {
+          found = true;
+        }
```

Comment:
pretty small - but you should be able to use `.some` here. It will return a boolean signifying if an element in the array was found. Will save you having to see set `found` manually:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

Reply:
👍  https://github.com/stellar/freighter/pull/407/commits/f6d1547be86d5c95f6fac60d5deb39b5502702d6